File: ClickServer/generatedJavaScript/com/pku/ss/egl/libs/PlayerControllerLib.js
define([],function(){
	egl.defineRUILibrary('com.pku.ss.egl.libs', 'PlayerControllerLib',
	{
		'eze$$fileName': 'com/pku/ss/egl/libs/PlayerControllerLib.egl',
		'eze$$runtimePropertiesFile': 'com.pku.ss.egl.libs.PlayerControllerLib',
			"constructor": function() {
				if(egl.com.pku.ss.egl.libs.PlayerControllerLib['$inst']) return egl.com.pku.ss.egl.libs.PlayerControllerLib['$inst'];
				egl.com.pku.ss.egl.libs.PlayerControllerLib['$inst']=this;
			}
			,
			"eze$$setEmpty": function() {
				this.p = new ();
			}
			,
			"eze$$setInitial": function() {
				if(egl.com.pku.ss.egl.libs.PlayerControllerLib['$inst']) return egl.com.pku.ss.egl.libs.PlayerControllerLib['$inst'];
				egl.com.pku.ss.egl.libs.PlayerControllerLib['$inst']=this;
				this.eze$$setEmpty();
			}
			,
			"eze$$getAnnotations": function() {
				if(this.annotations === undefined){
					this.annotations = {};
					this.annotations["XMLRootElement"] = new egl.eglx.xml.binding.annotation.XMLRootElement("PlayerControllerLib", null, false);
				}
				return this.annotations;
			}
			,
			"eze$$getFieldInfos": function() {
				if(this.fieldInfos === undefined){
					var annotations;
					this.fieldInfos = new Array();
				}
				return this.fieldInfos;
			}
			,
			"PlayMusic": function() {
				try {
					egl.com.pku.ss.egl.et.Player.playMusic();
				}
				catch ( eze$Temp1 )
				{
					{
						if (!(eze$Temp1 instanceof egl.eglx.lang.AnyException)) {
							eze$Temp1 = egl.makeExceptionFromCaughtObject(eze$Temp1);
						}
						var e = eze$Temp1;
						{
							egl.eglx.lang.SysLib.writeStderr(egl.checkNull(e).messageID);
							egl.eglx.lang.SysLib.writeStderr(egl.checkNull(e).message);
						}
					}
				}
			}
			,
			"PlayVideo": function() {
				try {
					egl.com.pku.ss.egl.et.Player.playVideo();
				}
				catch ( eze$Temp2 )
				{
					{
						if (!(eze$Temp2 instanceof egl.eglx.lang.AnyException)) {
							eze$Temp2 = egl.makeExceptionFromCaughtObject(eze$Temp2);
						}
						var e = eze$Temp2;
						{
							egl.eglx.lang.SysLib.writeStderr(egl.checkNull(e).message);
						}
					}
				}
			}
			,
			"getP": function() {
				return p;
			}
			,
			"setP": function(ezeValue) {
				this.p = ezeValue;
			}
			,
			"toString": function() {
				return "[PlayerControllerLib]";
			}
		}
	);
});

File: ClickServer/generatedJava/com/pku/ss/egl/libs/PlayerControllerLib.java
package com.pku.ss.egl.libs;
import org.eclipse.edt.javart.resources.*;
import org.eclipse.edt.javart.*;
import eglx.lang.SysLib;
import com.pku.ss.egl.et.Player;
import org.eclipse.edt.runtime.java.eglx.lang.EAny;
import java.lang.Object;
import eglx.lang.AnyException;
@SuppressWarnings("unused")
@javax.xml.bind.annotation.XmlRootElement(name="PlayerControllerLib")
public class PlayerControllerLib extends ExecutableBase {
	private static final long serialVersionUID = 10L;
	private transient Player p;
	public PlayerControllerLib() {
		super();
	}
	{
		ezeInitialize();
	}
	public void ezeInitialize() {
		p = new Player();
	}
	public Player getP() {
		return p;
	}
	public void setP(Player ezeValue) {
		p = ezeValue;
	}
	public void PlayMusic() {
		try {
			Player.playMusic();
		}
		catch ( java.lang.Exception eze$Temp1 ) {
			org.eclipse.edt.javart.util.JavartUtil.checkHandleable( eze$Temp1 );
			AnyException e;
			if ( eze$Temp1 instanceof AnyException ) {
				e = (AnyException)eze$Temp1;
			}
			else {
				e = org.eclipse.edt.javart.util.JavartUtil.makeEglException(eze$Temp1);
			}
			{
				SysLib.writeStderr(e.getMessageID());
				SysLib.writeStderr(e.getMessage());
			}
		}
	}
	public void PlayVideo() {
		try {
			Player.playVideo();
		}
		catch ( java.lang.Exception eze$Temp2 ) {
			org.eclipse.edt.javart.util.JavartUtil.checkHandleable( eze$Temp2 );
			AnyException e;
			if ( eze$Temp2 instanceof AnyException ) {
				e = (AnyException)eze$Temp2;
			}
			else {
				e = org.eclipse.edt.javart.util.JavartUtil.makeEglException(eze$Temp2);
			}
			{
				SysLib.writeStderr(e.getMessage());
			}
		}
	}
}
